### PR TITLE
Updated directlineclient to use sockets, added logging and extended timeout to 3 mins

### PR DESCRIPTION
**Pull request by gabog**
Relates to #243
This PR includes: 
- Added TestRequestTimeout (set to 180 secs) in appsettings.json and updated TestRunner to take it as a parameter and use it.
- Added logger to DirectLinetTestClient and added debug and trace messages to help us track failures.
- Updated appsettings log level to debug (for now) and fixed ScriptTestBase so it loads logging settings from appsettings.
- Excluded Python skills from ExpectReplies scenarios.
- Refactored DirectLineTestClient to use websockets instead of polling.
- Added retry logic to DirectLinetTestClient.StartConversation (if the app doesn't start within a minute or so, the whole conversation will end in limbo and the test will timout).
- Updated XUnit references in TranscriptTestRunner.csproj so VS doesn't think it is a unit test project

### CODE REVIEW

**Comment by gabog** [comment]
Found some issues, converting it to draft until I figure them out (any feedback is welcomed anyway)

**Comment by gabog** [comment]
@denscollo / @ceciliaavila, this branch is ready to merge, can you try it out with the pipeline and see if the timeout issues improve? 

Using azure app restart for testing may give weird results because the restarts are not immediate and may come in the middle of a test, so try it with the pipeline and let me know how it goes. 

Take a look at the code and let me know if you see anything strange but the changes to the DirectLineTestClient are pretty complex, we can jump on a call and I can explain what I did if you need more details. 

Thanks

**Review by ceciliaavila — CHANGES_REQUESTED** [review]
Just some details:
- Commented line in AppSettings.json.
- Missing update of the `EchoComposerTests` class to include the new arguments in the XUnitTestRunner instantiation and the rename of the _testScriptsFolder folder.

File: Tests/SkillFunctionalTests/appsettings.json [hunk 1]
```diff
@@ -2,11 +2,13 @@
   "Logging": {
     "LogLevel": {
       "Default": "Debug",
-      "System": "Information",
-      "Microsoft": "Information"
+      "Microsoft": "Warning"
     }
   },
 
+  // Test request timeout (in seconds)
```
[the remark this answers is not in the document]
Comment (gabog) [reply Tests/SkillFunctionalTests/appsettings.json]: Done, thanks!

**Review by ceciliaavila — APPROVED** [review]

**Pull request merged**